### PR TITLE
store retry intervals to resume from the same interval on restart

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: src/Simplex/Messaging/Agent/RetryInterval.hs [hunk 1]
```diff
@@ -28,6 +31,19 @@ data RetryInterval2 = RetryInterval2
...
+  { slowInterval :: Int,
+    fastInterval :: Int
+  }
+  deriving (Show)
+
+updateRetryInterval2 :: RI2State -> RetryInterval2 -> RetryInterval2
+updateRetryInterval2 RI2State {slowInterval, fastInterval} RetryInterval2 {riSlow, riFast} =
+  RetryInterval2
+    { riSlow = riSlow {initialInterval = slowInterval},
+      riFast = riFast {initialInterval = fastInterval}
```
Comment (spaced4ndy) [inline src/Simplex/Messaging/Agent/RetryInterval.hs:44]:
```suggestion
    { riSlow = riSlow {initialInterval = slowInterval, increaseAfter = 0},
      riFast = riFast {initialInterval = fastInterval, increaseAfter = 0}
```

File: src/Simplex/Messaging/Agent/Store/SQLite/Migrations/M20230320_retry_state.hs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+module Simplex.Messaging.Agent.Store.SQLite.Migrations.M20230320_retry_state where
+
+import Database.SQLite.Simple (Query)
+import Database.SQLite.Simple.QQ (sql)
+
+m20230320_retry_state :: Query
+m20230320_retry_state =
+  [sql|
+ALTER TABLE snd_messages ADD COLUMN retry_int_slow INTEGER NULL;
+ALTER TABLE snd_messages ADD COLUMN retry_int_fast INTEGER NULL;
```
Comment (spaced4ndy) [inline src/Simplex/Messaging/Agent/Store/SQLite/Migrations/M20230320_retry_state.hs:12]: I think convention of the existing code base is that nullability is not explicitly specified, why change?
```suggestion
ALTER TABLE snd_messages ADD COLUMN retry_int_slow INTEGER;
ALTER TABLE snd_messages ADD COLUMN retry_int_fast INTEGER;
```

File: src/Simplex/Messaging/Agent/Store/SQLite/Migrations/agent_schema.sql [hunk 3]
```diff
@@ -112,6 +112,8 @@ CREATE TABLE snd_messages(
   internal_id INTEGER NOT NULL,
   internal_hash BLOB NOT NULL,
   previous_msg_hash BLOB NOT NULL DEFAULT x'',
+  retry_int_slow INTEGER NULL,
+  retry_int_fast INTEGER NULL,
```
Comment (spaced4ndy) [inline src/Simplex/Messaging/Agent/Store/SQLite/Migrations/agent_schema.sql:116]:
```suggestion
  retry_int_slow INTEGER,
  retry_int_fast INTEGER,
```

File: src/Simplex/Messaging/Agent/RetryInterval.hs (hunk 1 above)
Comment (epoberezkin) [reply src/Simplex/Messaging/Agent/RetryInterval.hs:44]: I don't think it's needed

**Pull request merged**